pkg/cmd/kafka/update/update.go

```diff
@@ -0,0 +1,228 @@
+	cmd.Flags().StringVar(&opts.id, "id", "", opts.localizer.MustLocalize("kafka.update.flag.id"))
+	cmd.Flags().StringVar(&opts.owner, "owner", "", opts.localizer.MustLocalize("kafka.update.flag.owner"))
+	cmd.Flags().BoolVarP(&opts.skipConfirm, "yes", "y", false, opts.localizer.MustLocalize("kafka.update.flag.yes"))
+	cmd.Flags().StringVar(&opts.name, "name", "", opts.localizer.MustLocalize("kafka.update.flag.name"))
+
+	_ = cmd.MarkFlagRequired("owner")
+
+	_ = kafkacmdutil.RegisterNameFlagCompletionFunc(cmd, f)
+
+	return cmd
+}
+
+func run(opts *Options) error {
+	connection, err := opts.Connection(connection.DefaultConfigRequireMasAuth)
+	if err != nil {
+		return err
+	}
+
+	api := connection.API()
+
+	var kafkaInstance *kafkamgmtclient.KafkaRequest
+	ctx := context.Background()
+	if opts.name != "" {
+		kafkaInstance, _, err = kafka.GetKafkaByName(ctx, api.Kafka(), opts.name)
+		if err != nil {
+			return err
+		}
+	} else {
+		kafkaInstance, _, err = kafka.GetKafkaByID(ctx, api.Kafka(), opts.id)
+		if err != nil {
+			return err
+		}
+	}
+
+	if opts.owner == kafkaInstance.GetOwner() {
+		opts.logger.Info(opts.localizer.MustLocalize("kafka.update.log.info.sameOwnerNoChanges", localize.NewEntry("Owner", opts.owner), localize.NewEntry("InstanceName", kafkaInstance.GetName())))
+		return nil
+	}
+
+	updateObj := kafkamgmtclient.NewKafkaUpdateRequest(opts.owner)
+
+	// create a text block with a summary of what is being updated
+	updateSummary := generateUpdateSummary(reflect.ValueOf(*updateObj), reflect.ValueOf(*kafkaInstance))
+
+	opts.logger.Infof(`
+ %v 🗒️
+
+ %v`, color.Underline(color.Bold(opts.localizer.MustLocalize("kafka.update.summaryTitle"))), updateSummary)
+
+	if !opts.skipConfirm {
+		promptConfirm := survey.Confirm{
+			Message: opts.localizer.MustLocalize("kafka.update.confirmDialog.message", localize.NewEntry("Name", kafkaInstance.GetName())),
+		}
+		var confirmUpdate bool
+		if err = survey.AskOne(&promptConfirm, &confirmUpdate); err != nil {
+			return err
+		}
+		if !confirmUpdate {
+			opts.logger.Debug("User has chosen to not update Kafka instance")
+			return nil
+		}
+	}
+
+	spinner := opts.IO.NewSpinner()
+	spinner.Suffix = " " + opts.localizer.MustLocalize("kafka.update.log.info.updating", localize.NewEntry("Name", kafkaInstance.GetName()))
+	spinner.Start()
+
+	response, httpRes, err := api.Kafka().
+		UpdateKafkaById(context.Background(), kafkaInstance.GetId()).
+		KafkaUpdateRequest(*updateObj).
+		Execute()
+
+	spinner.Stop()
+
+	if apiError, ok := kas.GetAPIError(err); ok {
+		return errors.New(apiError.GetReason())
+	}
+
+	if err != nil {
+		return err
+	}
+	defer httpRes.Body.Close()
+
+	opts.logger.Infof(`
+
+%v`, opts.localizer.MustLocalize("kafka.update.log.info.updateSuccess", localize.NewEntry("Name", response.GetName())))
+
+	dump.PrintDataInFormat(opts.outputFormat, response, opts.IO.Out)
```

Comment:
[Optional] Changing owner of kafka that is used by you should probably remove it from config.

Reply:
I'm unsure. As an org admin I am still able to modify the Kafka instance after changing the ownership. Looking into the future, when more fields can be modified, an admin may want to make incremental changes.

```shell
$ rhoas kafka update --owner "new_owner" -y
...
$ rhoas kafka update --multi-az false
...
```

This is a general "update" command, which happens to have an owner field which can be updated in it. At some stage you can update multiple fields, then do we only "unuse" this instance when the owner field has been changed?

Reply:
Sorry. I wasn't aware of that. Yes. It is not about connectivity to kafka anymore but operating on it.